src/test/resources/epub3/navigation-document.feature

```diff
@@ -10,5 +10,95 @@ Feature: EPUB 3 Navigation Document
+    Then error RSC-005 is reported
+    And warning RSC-017 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document without a TOC nav, 2
+    When checking document 'invalid-nav-no-toc-002.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document with invalid hidden attribute
+    When checking document 'invalid-hidden-text.xhtml'
+    Then the following errors are reported
+      | RSC-005 | element "p" not allowed here; expected element "a" or "span" |
+      | RSC-005 | element "ol" not allowed yet; expected element "a" or "span" |
+      | RSC-005 | value of attribute "hidden" is invalid; must be equal to "" or "hidden" |
+      | RSC-005 | element "li" incomplete; missing required element "ol" |
+      | RSC-005 | Heading elements must contain text |
+      | RSC-005 | Heading elements must contain text |
+      | RSC-005 | Heading elements must contain text |
+      | RSC-005 | Spans within nav elements must contain text |
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has anchors that lack text
+    When checking document 'invalid-nav-empty-anchor.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has anchors that lack text
+    When checking document 'invalid-nav-empty-anchor-spans.xhtml'
+    Then the following errors are reported
+      | RSC-005 | Anchors within nav elements must contain text |
+      | RSC-005 | Spans within nav elements must contain text |
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has landmarks that lack epub:type
+    When checking document 'invalid-nav-landmarks-missing-anchor-type.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has too many landmarks elements
+    When checking document 'invalid-nav-landmarks-multiple.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has duplicate landmark elements
+    When checking document 'invalid-nav-landmarks-duplicates.xhtml'
+    Then error RSC-005 is reported
+    And error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has anchors that lack text
+    When checking document 'nav-landmarks-noduplicates.xhtml'
+    Then no errors or warnings are reported
+
+  Scenario: Nav document has multiple pagelist elements
+    When checking document 'invalid-nav-multiple-pagelists.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document has ordered list with no elements
+    When checking document 'invalid-nav-empty-ol.xhtml'
+    Then error RSC-005 is reported
+    And no other errors or warnings are reported
+
+  Scenario: Nav document lacks required headings
```

Comment:
having no heading is fine for the `toc`, `page-list`, and `landmarks` nav; the title could be explicit that this is a requirement for other navs.